src/zenml/models/v2/core/run_template.py

```diff
@@ -0,0 +1,385 @@
+
+from pydantic import Field
+
+from zenml.config.pipeline_spec import PipelineSpec
+from zenml.constants import STR_FIELD_MAX_LENGTH, TEXT_FIELD_MAX_LENGTH
+from zenml.enums import ExecutionStatus
+from zenml.models.v2.base.base import BaseUpdate
+from zenml.models.v2.base.scoped import (
+    WorkspaceScopedRequest,
+    WorkspaceScopedResponse,
+    WorkspaceScopedResponseBody,
+    WorkspaceScopedResponseMetadata,
+    WorkspaceScopedResponseResources,
+    WorkspaceScopedTaggableFilter,
+)
+from zenml.models.v2.core.code_reference import (
+    CodeReferenceResponse,
+)
+from zenml.models.v2.core.pipeline import PipelineResponse
+from zenml.models.v2.core.pipeline_build import (
+    PipelineBuildResponse,
+)
+from zenml.models.v2.core.pipeline_deployment import (
+    PipelineDeploymentResponse,
+)
+from zenml.models.v2.core.tag import TagResponse
+
+if TYPE_CHECKING:
+    from sqlalchemy.sql.elements import ColumnElement
+
+# ------------------ Request Model ------------------
+
+
+class RunTemplateRequest(WorkspaceScopedRequest):
+    """Request model for run templates."""
+
+    name: str = Field(
+        title="The name of the run template.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    description: Optional[str] = Field(
+        default=None,
+        title="The description of the run template.",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+
+    deployment_id: UUID = Field(
+        title="The deployment that should be the base of the created template."
+    )
+    tags: Optional[List[str]] = Field(
+        default=None,
+        title="Tags of the run template.",
+    )
+
+
+# ------------------ Update Model ------------------
+
+
+class RunTemplateUpdate(BaseUpdate):
+    """Run template update model."""
+
+    name: Optional[str] = Field(
+        default=None,
+        title="The name of the run template.",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    description: Optional[str] = Field(
+        default=None,
+        title="The description of the run template.",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    add_tags: Optional[List[str]] = Field(
+        default=None, title="New tags to add to the run template."
+    )
+    remove_tags: Optional[List[str]] = Field(
+        default=None, title="Tags to remove from the run template."
+    )
+
+
+# ------------------ Response Model ------------------
+
+
+class RunTemplateResponseBody(WorkspaceScopedResponseBody):
+    """Response body for run templates."""
+
+    runnable: bool = Field(
+        title="If a run can be started from the template.",
+    )
+    latest_run_id: Optional[UUID] = Field(
+        default=None,
+        title="The ID of the latest run of the run template.",
+    )
+    latest_run_status: Optional[ExecutionStatus] = Field(
+        default=None,
+        title="The status of the latest run of the run template.",
+    )
+    tags: List[TagResponse] = Field(
+        title="Tags associated with the run template.",
+    )
```

Comment:
Resources are the way here. It would work with the tags and since the status of pipeline runs are also in their body, we can move the latest run as well. 💪

Reply:
You mean having `latest_run` in the resources? I would agree with you, but right now a run has lots of stuff in it's body that we don't want to have in this response I think: `stack`, `pipeline`, `build`, `schedule`, `code_reference`. All stuff that should be in the run resources instead, but I don't want to change that as part of this PR and for backwards compatability.

Reply:
I moved the tags though, for templates, pipelines and runs (all the entities where I introduced them).